core/src/main/scala/kafka/coordinator/ConsumerCoordinator.scala

```diff
@@ -210,7 +210,9 @@ class ConsumerCoordinator(val brokerId: Int,
             responseCallback(Errors.UNKNOWN_CONSUMER_ID.code)
           } else if (!group.has(consumerId)) {
             responseCallback(Errors.UNKNOWN_CONSUMER_ID.code)
-          } else if (generationId != group.generationId || !group.is(Stable)) {
+          } else if (!group.is(Stable)) {
+            responseCallback(Errors.REBALANCE_IN_PROGRESS.code)
```

Comment:
I think I got the ordering wrong here. We probably should check for generation id mismatch before validating group stability to give a more accurate error code back to the user.

I'll update the PR.